be/src/vec/functions/function_jsonb.cpp

```diff
@@ -503,6 +508,93 @@ struct JsonbExtractStringImpl {
+                                r_raw, r_size, is_invalid_json_path);
+            } else { // will make array string to user
+                writer->reset();
+                writer->writeStartArray();
+                for (size_t pi = 0; pi < rdata_columns.size(); ++pi) {
+                    const ColumnString* path_col = rdata_columns[pi];
+                    const ColumnString::Chars& rdata = path_col->get_chars();
+                    const ColumnString::Offsets& roffsets = path_col->get_offsets();
+                    size_t r_off = roffsets[index_check_const(i, path_const[pi]) - 1];
+                    size_t r_size = roffsets[index_check_const(i, path_const[pi])] - r_off;
+                    const char* r_raw = reinterpret_cast<const char*>(&rdata[r_off]);
+                    String path(r_raw, r_size);
+                    // doc is NOT necessary to be deleted since JsonbDocument will not allocate memory
+                    JsonbDocument* doc = JsonbDocument::createDocument(l_raw, l_size);
+                    if (UNLIKELY(!doc || !doc->getValue())) {
+                        writer->writeNull();
+                        continue;
+                    }
+                    // value is NOT necessary to be deleted since JsonbValue will not allocate memory
+                    JsonbValue* value =
+                            doc->getValue()->findPath(r_raw, r_size, is_invalid_json_path, nullptr);
+                    // if not valid json path , should return error message to user
+                    if (is_invalid_json_path) {
+                        return Status::InvalidArgument(
+                                "Json path error: {} for value: {}",
+                                JsonbErrMsg::getErrMsg(JsonbErrType::E_INVALID_JSON_PATH),
+                                std::string_view(reinterpret_cast<const char*>(rdata.data()),
+                                                 rdata.size()));
+                    }
+                    if (UNLIKELY(!value)) {
+                        writer->writeNull();
+                        continue;
+                    }
+                    writer->writeValue(value);
+                }
+                writer->writeEndArray();
+                if constexpr (std::is_same_v<DataTypeJsonb, ReturnType>) {
+                    StringOP::push_value_string(std::string_view(writer->getOutput()->getBuffer(),
+                                                                 writer->getOutput()->getSize()),
+                                                i, res_data, res_offsets);
+                } else {
+                    // transform jsonb to json
+                    if (!formater) {
+                        formater.reset(new JsonbToJson());
+                    }
+                    StringOP::push_value_string(
+                            formater->to_json_string(writer->getOutput()->getBuffer(),
```

Comment:
In this branch, writer is not initialized since ReturnType is not Jsonb. In fact, I think format to string is not necessary, since planner will automaticlly add cast from JSONB to  String if necessary.